File: test_api.js
const http = require('http');

const loginData = JSON.stringify({
  email: 'admin@test.com',
  password: 'password123'
});

const req = http.request({
  hostname: '127.0.0.1',
  port: 5000,
  path: '/api/v1/auth/login',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': loginData.length
  }
}, res => {
  let data = '';
  res.on('data', chunk => data += chunk);
  res.on('end', () => {
    try {
      const parsed = JSON.parse(data);
      const token = parsed.data.accessToken || parsed.data.token;
      console.log('Token:', token ? 'Found' : 'Not found');
      
      // Fetch users
      http.get('http://127.0.0.1:5000/api/v1/users', { headers: { 'Authorization': 'Bearer ' + token } }, res2 => {
        let uData = '';
        res2.on('data', chunk => uData += chunk);
        res2.on('end', () => {
          console.log('Users response structure:');
          try {
            const parsedU = JSON.parse(uData);
            console.log(Object.keys(parsedU));
            if (parsedU.data) {
                console.log('Data keys:', Object.keys(parsedU.data));
                if(Array.isArray(parsedU.data)) console.log('Data is an array');
            }
          } catch(e) { console.log('UData:', uData.substring(0, 100)) }
        });
      });
      
      // Fetch admin courses
      http.get('http://127.0.0.1:5000/api/v1/admin/courses', { headers: { 'Authorization': 'Bearer ' + token } }, res3 => {
        let cData = '';
        res3.on('data', chunk => cData += chunk);
        res3.on('end', () => {
          console.log('Admin courses response structure:');
          try {
            const parsedC = JSON.parse(cData);
            console.log(Object.keys(parsedC));
            if (parsedC.data) {
                console.log('Data keys:', Object.keys(parsedC.data));
                if(Array.isArray(parsedC.data)) console.log('Data is an array');
            }
          } catch(e) { console.log('CData:', cData.substring(0, 100)) }
        });
      });
    } catch (e) {
      console.log('Login parse error', e);
      console.log('Raw:', data);
    }
  });
});

req.write(loginData);
req.end();

File: test_api.dart
import 'package:dio/dio.dart';
import 'dart:convert';

void main() async {
  final dio = Dio();
  // We need a login token to test the API.
  // We can login as the instructor.
  try {
    final loginRes = await dio.post(
      'http://localhost:5000/api/v1/auth/login',
      data: {'email': 'instructor@example.com', 'password': 'password123'},
    );
    final token = loginRes.data['data']['accessToken'];
    print('Logged in, token: \$token');

    dio.options.headers['Authorization'] = 'Bearer \$token';

    // Fetch instructor courses
    final coursesRes = await dio.get('http://localhost:5000/api/v1/courses/instructor');
    final courses = coursesRes.data['data']['courses'] as List;
    if (courses.isEmpty) {
      print('No courses');
      return;
    }
    final courseId = courses.first['id'] ?? courses.first['_id'];
    
    // Fetch quizzes for course
    final quizzesRes = await dio.get('http://localhost:5000/api/v1/courses/\$courseId/quizzes/instructor');
    final quizzes = quizzesRes.data['data']['quizzes'] as List;
    if (quizzes.isEmpty) {
      print('No quizzes');
      return;
    }
    final quizId = quizzes.first['id'] ?? quizzes.first['_id'];
    
    // Get detailed quiz
    final detailRes = await dio.get('http://localhost:5000/api/v1/quizzes/\$quizId/instructor');
    print(jsonEncode(detailRes.data));

  } catch (e) {
    if (e is DioException) {
      print('DioError: \${e.response?.data}');
    } else {
      print('Error: \$e');
    }
  }
}
